crates/bevy_reflect/src/type_name.rs

```diff
@@ -1,15 +1,58 @@
 use crate::utility::GenericTypeNameCell;
 
-/// Provide the name of the type as string.
-pub trait TypeName {
+/// Provide the name of the type as a string slice.
+///
+/// This is a stable alternative to [`std::any::type_name`] whose output isn't guarentee
+/// and may change between versions of the compiler.
+///
+/// This trait may be derived via [`#[derive(TypeName)]`][bevy_reflect_derive::TypeName].
+///
+/// ## Manual implementation
+///
+/// For some reason you may need to manually implement [`TypeName`].
```

Comment:
Nit: This might sound a bit nicer
```suggestion
/// If you need to manually implement [`TypeName`], it's recommended to follow
/// the example below (unless specifying a custom name).
```